packages/sdk-react/src/components/receivables/InvoiceDetails/ExistingInvoiceDetails/components/OverviewTabPanel.tsx

```diff
@@ -75,6 +90,58 @@ export const OverviewTabPanel = ({
       }
     );
 
+  const { data: receivable } = useReceivableById(invoice.id);
+
+  const isInvoiceResponsePayload = (
+    receivable: components['schemas']['ReceivableResponse'] | undefined
+  ): receivable is components['schemas']['InvoiceResponsePayload'] =>
+    (receivable as components['schemas']['InvoiceResponsePayload'])
+      .related_documents !== undefined;
+
+  const creditNoteIds = isInvoiceResponsePayload(receivable)
+    ? receivable.related_documents.credit_note_ids
+    : [];
+
+  const {
+    data: creditNoteQuery,
+    isLoading: isCreditNoteLoading,
+    error: creditNoteError,
+  } = useReceivables({
+    id__in: creditNoteIds,
+    type: 'credit_note',
+  });
+
+  const transformCreditNotes = (
+    creditNotes: components['schemas']['InvoiceResponsePayload'][]
+  ): TransformCreditNotes[] => {
+    if (!creditNotes) return [];
+
+    return creditNotes.map((creditNote) => {
+      const issueDate = creditNote.issue_date
+        ? new Date(creditNote.issue_date)
+        : null;
+
+      const formattedDate = issueDate
+        ? issueDate.toLocaleDateString('en-GB').replace(/\//g, '.')
+        : t(i18n)`Unknown date`;
+
+      const authorName =
```

Comment:
You are probably looking for the [`getCounterpartName(...)`](https://github.com/team-monite/monite-sdk/blob/b47e6d52d731f41183a1a3eebc9d34a42e0d9234/packages/sdk-react/src/components/counterparts/helpers.ts#L43) function.

Reply:
Are we sure? I can create separate utils, if we want
<img width="1194" alt="Screenshot 2024-08-12 at 14 17 32" src="https://github.com/user-attachments/assets/c7044d08-8b4d-4342-af2c-237e2215662d">

Reply:
Fair enough, good!